src/Stratis.Bitcoin.Features.Consensus/ConsensusController.cs

```diff
@@ -46,5 +73,31 @@ public uint256 GetBlockHash(int height)
             ChainedHeader block = this.Chain.GetBlock(height);
             return block == null || block.Height > bestBlock.Height ? null : block.HashBlock;
         }
+
+        /// <summary>
+        /// Gets the hash of the block at the given height.
+        /// API implementation of RPC call.
+        /// </summary>
+        /// <param name="request">A <see cref="GetBlockHashRequestModel"/> request containing the height.</param>
+        /// <returns>Json formatted <see cref="uint256"/> hash of the block at the given height. Returns <see cref="IActionResult"/> formatted error if fails.</returns>
+        [Route("getblockhash")]
+        [HttpGet]
+        public IActionResult GetBlockHash(GetBlockHashRequestModel request)
```

Comment:
Why a special request model is created? We can use a single param- height. 
Also why height is string and not int?